src/main/java/com/icthh/xm/tmf/ms/communication/service/KafkaHandelMessageService.java

```diff
@@ -0,0 +1,154 @@
+import com.icthh.xm.tmf.ms.communication.web.api.model.CommunicationMessage;
+import com.icthh.xm.tmf.ms.communication.web.api.model.CommunicationRequestCharacteristic;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.StopWatch;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.icthh.xm.tmf.ms.communication.rules.ttl.TTLRule.MESSAGE_RECEIVED_BY_CHANNEL_TIMESTAMP;
+import static org.apache.commons.lang3.StringUtils.unwrap;
+import static org.springframework.kafka.support.KafkaHeaders.ACKNOWLEDGMENT;
+
+@Slf4j
+@Service
+@ConditionalOnProperty("application.stream-binding-enabled")
+public class KafkaHandelMessageService {
+
+    private static final int MICROSECONDS_IN_SEC = 1_000_000;
+    public static final String TENANT_NAME = "TENANT.NAME";
+    public static final String XM = "XM";
+
+
+    private final int pauseBetweenSends;
+    private final LepKafkaMessageHandler lepMessageHandler;
+    private final long kafkaReadSleepTimeout;
+    private final MessageHandlerService messageHandlerService;
+    private final ObjectMapper objectMapper;
+    private volatile AtomicLong nextScheduledTime;
+
+    private ScheduledExecutorService scheduledExecutorService;
+
+    @Autowired
+    public KafkaHandelMessageService(ApplicationProperties applicationProperties,
+                                     MessageHandlerService messageHandlerService,
+                                     LepKafkaMessageHandler lepMessageHandler,
+                                     ObjectMapper objectMapper) {
+        this.messageHandlerService = messageHandlerService;
+        this.lepMessageHandler = lepMessageHandler;
+        this.objectMapper = objectMapper;
+
+        pauseBetweenSends = MICROSECONDS_IN_SEC / applicationProperties.getKafka().getRateLimit();
+        int processingPoolSize = applicationProperties.getKafka().getPoolSize();
+        nextScheduledTime = new AtomicLong((System.nanoTime() / 1000) + pauseBetweenSends);
+        scheduledExecutorService = Executors.newScheduledThreadPool(processingPoolSize);
+        kafkaReadSleepTimeout = processingPoolSize * pauseBetweenSends * 2;
+    }
+
+    public void handle(Message<?> message) {
+
+        long delay = nextScheduledTime.getAndAdd(pauseBetweenSends) - System.nanoTime() / 1000;
```

Comment:
Seems rate limiting is an additional feature. Should not we use some feature flag (disabled by default) to preserve old logic without rate limiting?